### PR TITLE
#406 usage aggregation logic fix

### LINKED ISSUE #406: Item usage not aggregated across batches correctly
Build Number: 30

Description: It looks like item.dailyUsage() is summing the dailyUsage for each batch, which would over-estimate usage when there are multiple batches. If daily usage for batch 1 is 15, and daily usage for batch 2 is also 15, it looks like item.dailyUsage() would return 30. The spreadsheet example linked in issue #405 also includes something about this. It seems like the correct approach to get the total daily usage for an item across batches would be to sum up the total usage for all batches and the number of days each batch was used, then do the division in Item.

Comments: Happy to take a crack at this ticket or #405, but I don't have a server I could test on at the moment..

### PR DESCRIPTION
**Pull request by Chris-Petty**
Refactors dailyUsage as described in #406.

fixes #406

### CODE REVIEW

**Comment by Chris-Petty** [comment]
@gkaplan You might like to have a pass at this 💃

File: src/database/DataTypes/ItemBatch.js [hunk 1]
```diff
@@ -16,37 +12,18 @@ export class ItemBatch extends Realm.Object {
...
-    const currentDate = new Date();
-    let usagePeriod = millisecondsToDays(USAGE_PERIOD_MILLISECONDS);
-    if (transactionBatches.length === this.transactionBatches.length) {
-      // This item batch has no transaction batches older than the usage period constant,
-      // use the actual amount of time it has been around for as the usage period
-      usagePeriod = millisecondsToDays(currentDate.getTime() - this.addedDate.getTime());
-    }
-    const dailyUsage = usagePeriod ? totalUsage / usagePeriod : 0;
-    return dailyUsage;
+    throw new Error('ItemBatch.dailyUsage() is a deprecated function');
```
Comment (Chris-Petty) [inline src/database/DataTypes/ItemBatch.js]: I should probably remove this method before PR.

File: src/database/DataTypes/Item.js [hunk 2]
```diff
@@ -11,19 +13,35 @@ export class Item extends Realm.Object {
...
+      const batchAddedDate = batch.addedDate;
+      itemAddedDate = batchAddedDate < itemAddedDate ? batchAddedDate : itemAddedDate;
+    });
+    return itemAddedDate;
   }
 
   get earliestExpiringBatch() {
     if (this.batches.length === 0) return null;
-    let earliestBatch = this.batches.find((batch) => batch.totalQuantity > 0);
+    let earliestBatch = this.batches.find(batch => batch.totalQuantity > 0);
```
Comment (Chris-Petty) [inline src/database/DataTypes/Item.js]: I've run prettier locally which culled the parentheses. We don't have an eslint rule against not parenthesising single params in a arrow function and I'm becoming indifferent towards enforcing this coding style.

Comment (andreievg) [inline src/database/DataTypes/Item.js]: looks tidier to me

**Pull request merged**